types/devicefarmer__stf-syrup/index.d.ts

```diff
@@ -0,0 +1,55 @@
+import Bluebird = require("bluebird");
+
+type extractDesRet<RetT> = RetT extends SyrupI<never, never, infer RetX> ? RetX
+    : unknown;
+type extractBluebirdReturnR<RetT> = RetT extends Bluebird<infer RetX> ? RetX
+    : RetT;
+declare class SyrupI<
+    OptionsT extends object = any, // TODO: find a way to remove any. Maybe we union all the options that are needed for each dependency?
+    DepsT extends SyrupI[] = [],
+    RetT = unknown,
+    DepsRetsT extends unknown[] = [], // TODO: maybe we can extract DepsRetsT somehow?
+> {
+    constructor(options: OptionsT | null);
+    define<
+        BodyT extends (options: OptionsT, ...deps: DepsRetsT) => unknown,
+    >(
+        body: BodyT,
+    ): SyrupI<
+        OptionsT,
+        DepsT,
+        extractBluebirdReturnR<ReturnType<typeof body>>,
+        DepsRetsT
+    >;
+    dependency<DepT extends SyrupI<never, never>>(
+        dep: DepT,
+    ): SyrupI<
+        OptionsT,
+        [...DepsT, DepT],
+        RetT,
+        [...DepsRetsT, extractDesRet<DepT>]
+    >;
+    consume<OptionsT>(
+        overrides: OptionsT,
+    ): Bluebird<RetT>;
+    invoke(overrides: OptionsT, ...args: DepsT[]): RetT;
+}
+
+declare function ParallelSyrup(
+    options?: object,
```

Comment:
It might be right for your use case, but `object` is typically an indication that something isn't right. `object` means "things that are `typeof === "object"`, which means both objects and null.

What was the intent for these? Would `unknown` suffice for these option parameters?

Reply:
The library accepts basically anything as options, and just passes it as the first argument to the module, but I wanted to specify that it should be some object